File: 0621-task-scheduler/0621-task-scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] cnt = new int[26];
        int most = 0, maximum = 0;

        for (char c: tasks) {
            cnt[c - 'A']++;
            if (maximum == cnt[c - 'A']) {
                most++;
            } else if (maximum < cnt[c - 'A']) {
                maximum = cnt[c - 'A'];
                most = 1;
            }
        }

        int result = (maximum - 1) * (n + 1) + most;
        return Math.max(result, tasks.length);
    }
}

File: 0621-task-scheduler/0621-task-scheduler.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        cnt = Counter(tasks)
        cnt_sorted = [x[1] for x in cnt.most_common()]
        most = 0

        for i in range(len(cnt_sorted)):
            if cnt_sorted[i] == cnt_sorted[0]:
                most += 1
            else:
                break

        result = (cnt_sorted[0] - 1) * (n + 1) + most

        return max(result, len(tasks))